### PR TITLE
Make Display more generic and remove lot of redundant code

### PR DESCRIPTION
**Pull request by peckpeck**
As suggested in #121

### CODE REVIEW

**Comment by caemor** [comment]
Thanks for the great PR(s)! :partying_face: 

I just took a quick glance and it looks mostly very good. 
Just to make it easier to review and to make it easier to assert that nothing changed for the different displays: Could you move a few of the testcases respective 'graphics' file to the 'mod' files?

File: src/epd1in54/graphics.rs [hunk 1]
```diff
@@ -1,164 +0,0 @@
...
-        let buffer = display.buffer();
-
-        assert_eq!(buffer[0], Color::Black.get_byte_value());
-
-        for &byte in buffer.iter().skip(1) {
-            assert_eq!(byte, DEFAULT_BACKGROUND_COLOR.get_byte_value());
-        }
-    }
-
-    #[test]
```
Comment (caemor) [inline src/epd1in54/graphics.rs:125]: E.g. keep one of these testcase in the epd1in54/mod.rs file

File: src/epd1in54/graphics.rs [hunk 2]
```diff
@@ -1,164 +0,0 @@
...
-
-        extern crate std;
-        std::println!("{:?}", buffer);
-
-        assert_eq!(buffer[0], Color::Black.get_byte_value());
-
-        for &byte in buffer.iter().skip(1) {
-            assert_eq!(byte, DEFAULT_BACKGROUND_COLOR.get_byte_value());
-        }
-    }
```
Comment (caemor) [inline src/epd1in54/graphics.rs:143]: E.g. keep one of these testcase in the epd1in54/mod.rs file

**Comment by peckpeck** [comment]
Fixed, and also fixed a bug that made Tricolor case fail

**Comment by peckpeck** [comment]
Any news on this ?

File: src/graphics.rs [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
 
-use crate::buffer_len;
-use crate::color::{Color, OctColor, TriColor};
-use embedded_graphics_core::pixelcolor::BinaryColor;
+use crate::color::{ColorType, TriColor};
+use core::marker::PhantomData;
 use embedded_graphics_core::prelude::*;
 
-/// Displayrotation
+/// Display rotation, only 90° increment supported
```
Comment (caemor) [inline src/graphics.rs:7]:
```suggestion
/// Display rotation, only 90° increments supported
```

**Pull request merged**